sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/QuerySamples.cs

```diff
@@ -0,0 +1,235 @@
+        /// </summary>
+        public async Task RunSamplesAsync(TimeSeriesInsightsClient client, DeviceClient deviceClient)
+        {
+            // Figure out what keys make up the Time Series Id
+            TimeSeriesModelSettings modelSettings = await client.ModelSettings.GetAsync().ConfigureAwait(false);
+
+            int numOfIdKeys = modelSettings.TimeSeriesIdProperties.Count;
+
+            // Create a Time Series Id where the number of keys that make up the Time Series Id is fetched from Model Settings
+            var id = new List<string>();
+            for (int i = 0; i < numOfIdKeys; i++)
+            {
+                id.Add(Guid.NewGuid().ToString());
+            }
+
+            TimeSeriesId tsId = numOfIdKeys switch
+            {
+                1 => new TimeSeriesId(id[0]),
+                2 => new TimeSeriesId(id[0], id[1]),
+                3 => new TimeSeriesId(id[0], id[1], id[2]),
+                _ => throw new Exception($"Invalid number of Time Series Insights Id properties."),
+            };
+
+            // In order to query for data, let's first send events to the IoT Hub
+            await SendEventsToIotHubAsync(deviceClient, tsId, modelSettings.TimeSeriesIdProperties.ToArray()).ConfigureAwait(false);
+
+            // Sleeping for a few seconds to allow data to fully propagate in the Time Series Insights service
+            Thread.Sleep(TimeSpan.FromSeconds(3));
+
+            await RunQueryEventsSample(client, tsId).ConfigureAwait(false);
+
+            await RunQuerySeriesSample(client, tsId).ConfigureAwait(false);
+
+            await RunQueryAggregateSeriesSample(client, tsId).ConfigureAwait(false);
+        }
+
+        private async Task RunQueryAggregateSeriesSample(TimeSeriesInsightsClient client, TimeSeriesId tsId)
+        {
+            #region Snippet:TimeSeriesInsightsSampleQueryAggregateSeriesWithNumericVariable
+            Console.WriteLine("\n\nQuery for the average temperature over the past 30 seconds, in 2-second time slots.\n");
+
+            var numericVariable = new NumericVariable(
+                new TimeSeriesExpression("$event.Temperature"),
+                new TimeSeriesExpression("avg($value)"));
+
+            var requestOptions = new QueryAggregateSeriesRequestOptions();
+            requestOptions.InlineVariables["Temperature"] = numericVariable;
+            requestOptions.ProjectedVariables.Add("Temperature");
+
+            QueryAnalyzer queryAggregateSeriesAnalyzer = client.Query.CreateAggregateSeriesQueryAnalyzer(
+                tsId,
+                TimeSpan.FromSeconds(2),
+                TimeSpan.FromSeconds(30),
+                null,
+                requestOptions);
+
+            await foreach (TimeSeriesPoint point in queryAggregateSeriesAnalyzer.GetResultsAsync())
+            {
+                double? averageTemperature = (double?)point.GetValue("Temperature");
+                if (averageTemperature != null)
```

Comment:
Just to understand in which scenarios can it be null?

Reply:
Let's say we are looking at a 1 day graph. This 1 day graph might not have values at all time slots (maybe for the first 10 hours, there is no temperature value, but for the next 10 there). So this is when it can be null.

Think of a graph that is not complete from left to right, rather chopped up.

Reply:
Thanks.